src/hotspot/share/opto/inlinetypenode.cpp

```diff
@@ -433,6 +433,12 @@ void InlineTypeNode::load(GraphKit* kit, Node* base, Node* ptr, ciInstanceKlass*
           ft = con_type->inline_klass();
           null_free = true;
         }
+        BasicType bt = con_type->basic_type();
+        int vec_len = field->secondary_fields_count();
+        if (field->is_multifield_base() &&
+          Matcher::match_rule_supported_vector(VectorNode::replicate_opcode(bt), vec_len, bt)) {
+          value = kit->gvn().transform(VectorNode::scalar2vector(value, vec_len, Type::get_const_type(field->type()), false));
```

Comment:
If the oop is not the default oop, the multifield values maybe different from the multifield base, right?

Reply:
Correct, its handles loading a multi-field value from a constant oop associated with payload which is created only when payload class is uninitialized. 

https://github.com/openjdk/valhalla/pull/833/files#diff-ba9e2d10a50a01316946660ec9f68321eb864fd9c815616c10abbec39360efe5R1692